addons/web_editor/controllers/main.py

```diff
@@ -191,7 +191,8 @@ def add_data(self, name, data, is_image, quality=0, width=0, height=0, res_id=Fa
             try:
                 data = tools.image_process(data, size=(width, height), quality=quality, verify_resolution=True)
                 mimetype = guess_mimetype(b64decode(data))
-                if mimetype not in SUPPORTED_IMAGE_MIMETYPES:
+                img = ImageProcess(data, verify_resolution=False)
+                if not img.image or mimetype not in SUPPORTED_IMAGE_MIMETYPES:
```

Comment:
I took a look at `ImageProcess` to see in which cases `img.image` would be falsy and I found that it could only be if `data[:1] in (b'P', 'P')`, which if I'm not mistaken is simply checking that the file starts with `<` as in `<?xml`. Seems like that should be enough for our purposes here.

```suggestion
                is_svg = data[:1] in (b'P', 'P')`
                if is_svg or mimetype not in SUPPORTED_IMAGE_MIMETYPES:
```

Reply:
Hello @Zynton 
Requested changes are done.
Thanks!

Reply:
Hi! Thanks for making the changes.

I took a deeper look and I'm actually confused as to the general approach here. I got curious as to why the pre-existing call to `guess_mimetype` wasn't enough. I assumed it must do something similar to detect svg as what we are doing here, though probably more involved. And sure enough, I uploaded a svg with `.jpg` as file extension and it still correctly returned `image/svg+xml`.

You say above "an unsupported mimetype such as 'svg'", but then I look at the definition of `SUPPORTED_IMAGE_MIMETYPES` and see `image/svg+xml` in it:
```py
SUPPORTED_IMAGE_MIMETYPES = ['image/gif', 'image/jpe', 'image/jpeg', 'image/jpg', 'image/png', 'image/svg+xml']
```

So in fact `svg` is supposed to be supported? If not, then either it should be removed from `SUPPORTED_IMAGE_MIMETYPES` or we should simply do `if mimetype == 'image/svg+xml' or mimetype not in SUPPORTED_IMAGE_MIMETYPES:`. Either way, no need to try and detect svgs: `guess_mimetype` does it correctly already!

Reply:
Hello @Zynton 
Requested changes are done.
I have added ```if mimetype == 'image/svg+xml' or mimetype not in SUPPORTED_IMAGE_MIMETYPES:``` as per your suggestion.
I did not remove ```image/svg+xml``` from ```SUPPORTED_IMAGE_MIMETYPES``` because it can be used elsewhere.
Thanks!

Reply:
Thanks. I was actually asking a question. So you confirm svg were in fact not supposed to be supported? Where does that information come from?

Reply:
Hello @Zynton 
User can upload the .svg file,
The main problem is that user is uploading corrupt file. that's why error is generated.
So here, Problem is not for mimetype, problem is that file is corrupted.
IMO, we can use ```ImageProcess``` for it.
If you have any suggestions, please let me know.
Thanks!

Reply:
Mentioned this PR in https://www.odoo.com/web#id=3478929&cids=1&model=project.task&view_type=form we'll probably investigate further when we get to it. Unless we have a more legitimate flow than "create a svg image which is a jpeg" :wink: